streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java

```diff
@@ -17,73 +17,72 @@
 
 package org.apache.kafka.streams.kstream;
 
+import org.apache.kafka.common.annotation.InterfaceStability;
 import org.apache.kafka.common.serialization.Serde;
 import org.apache.kafka.streams.KeyValue;
 
 /**
- * KTable is an abstraction of a change log stream from a primary-keyed table.
+ * KTable is an abstraction of a <i>changelog stream</i> from a primary-keyed table.
  *
  * @param <K> Type of primary keys
  * @param <V> Type of value changes
  */
+@InterfaceStability.Unstable
 public interface KTable<K, V> {
 
     /**
-     * Creates a new instance of KTable consists of all elements of this stream which satisfy a predicate
+     * Create a new instance of {@link KTable} consists of all elements of this stream which satisfy a predicate.
      *
-     * @param predicate the instance of Predicate
-     * @return the instance of KTable with only those elements that satisfy the predicate
+     * @param predicate     the instance of {@link Predicate}
      */
     KTable<K, V> filter(Predicate<K, V> predicate);
 
     /**
-     * Creates a new instance of KTable consists all elements of this stream which do not satisfy a predicate
+     * Create a new instance of {@link KTable} consists all elements of this stream which do not satisfy a predicate
```

Comment:
It's a bit tricky. `filterNot` is probably better from a familiarity perspective: Scala has it and Java 8 has `filter` (so one can easily figure out that `filterNot` is the opposite of `filter`). However, both `filter` and `filterNot` are a bit confusing for a new user (it would be reasonable to think that the `filter` predicate would indicate which items are to be excluded from the result). `filterOut` seems a bit better in that respect, but I am not sure the benefit outweighs the familiarity point.